### ISSUE TITLE
Porting editor to Qt6

### ISSUE THREAD

**Issue by algts** [horde3d/Horde3D#217]
Although editor is not actively developed, it has to be able to compile with modern versions of Qt.
Main issues with porting to Qt6 are:

- QtOpenGL (and QGLWidget) are deprecated and removed from Qt6. Therefore main gl widget should be ported to QOpenGLWidget or some similar widget.
- Gizmos are drawn using OpenGL 1. Although we can still work with that, it is probably better to use a more modern approach. Inventing gizmos one more time is imho unneccessary, it is probably better to integrate projects like [im3d](https://github.com/john-chapman/im3d) or [imguizmo](https://github.com/CedricGuillemet/ImGuizmo).

**Comment by gwald** [comment]
I would suggest the less dependencies the better, ie using H3D's internal Geometry data, but I'm unfamiliar with that TBH.

**Comment by algts** [comment]
I have already ported editor to Qt6 and almost integrated im3d. Everything almost works, some issues here and there, but overall its good. You can see the progress in my repo in editor_upgrade branch.

**Comment by gwald** [comment]
Cool :)
When it's ready for testing, put up a build :+1: 
Also, I was wondering about animation, it looks like the scene editor doesn't know about them?
It would be good if anims could be included, like a reference from the model, I'm not talking about a full Unity3d editor, just so the code can be made aware of anim file and to model file link.

**Comment by algts** [comment]
@gwald Yes, currently editor knows nothing about animations, but it provides lua bindings, so you can write a mini lua script that will launch the animation for your model (not the best approach, I know :) ).

**Comment by gwald** [comment]
I think I like adding extra information using group nodes in the editor :) ie:

 <Group name="anim" tz="12.85820" tx="0.00000" ty="0.00000">
        <Group name="model1" tz="0.00000" tx="0.00000" ty="0.00000">
            <Group name="/animations/model/run.anim" tz="0.00000" tx="0.00000" ty="0.00000"/>
            <Group name="/animations/model/walk.anim" tz="0.00000" tx="0.00000" ty="0.00000"/>
        </Group>
    </Group>

I have to test this approach.

What was the idea behind "Add Attachment"? It crashes for me, maybe remove it if not used.

The lua intellisesne feature isn't working for me 100%, I type Horde3d and the dot, I get all the listing, but it goes away when I type more.


When the mouse pointer is over the coordinate widget, it would be good if the axis arrow could change color (yellow?) or flash?, something that shows that I wont miss when I click on it :)

**Comment by algts** [comment]
"Add attachment" was used for game engine integration, so that game engine components get used/updated via horde3d.

I'll look into issue with lua code completion.

As for gizmos, axis highlighting would be there in the upgraded version :)

**Comment by gwald** [comment]
Cool thanks :)
Happy to test it when it's ready.
We've read the text of the old Horde3D Editor video here, it's not serious, but hopefully it's helpful and brings awareness: https://youtu.be/Sgko0Ak3JrA

**Comment by gwald** [comment]
The drag and drop in the scene tree create copies, I was expecting it to be moved, like in a local file browser, but I understand that would be the case most often, so that's okay.

I don't think there's a way of moving items tho?
Maybe add right click to move items?

**Comment by algts** [comment]
Porting to Qt6 seems to be done. There may be bugs here and there, but should be working at least.
Here is the windows build: https://www.sendspace.com/file/bkgcki

If you encounter some bugs, please post here, I'm not closing the ticket until all major bugs are gone.
P.S.: Pull request with source code is there: #221

**Comment by gwald** [comment]
thanks, I dont have a project which uses the editor yet, but I'll give it test.

**Comment by gwald** [comment]
I've been using it to create a simple scene, it works great.
I really like the extra handles on the objects

**Comment by gwald** [comment]
I uploaded your QT6 editor here but with the new fixed converter executable, (removes white spaces from the DAE file).
https://archive.org/details/algts-horde3d-editor-win-x64-march-2023
